src/EditorFeatures/Core/Host/IStreamingFindReferencesPresenter.cs

```diff
@@ -65,12 +66,17 @@ internal static class IStreamingFindUsagesPresenterExtensions
             ImmutableArray<DefinitionItem> items,
             CancellationToken cancellationToken)
         {
-            // Can only navigate or present items on UI thread.
-            await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
❔ How do we verify that all down-stream code that depends on the main thread switches to it?

Reply:
so what i tried to do (now that these paths are actually async) is to push the main thread switches close to the point that we actually need to be on the UI thread (for example, calling into VS apis, or casting to VS types).

Reply:
also, i manually walked all the calls to the final points to try to ensure eveyrthing was right.  and i manually validated all scenarios that touch these.